xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp

```diff
@@ -1553,8 +1553,19 @@ bool CAMLCodec::OpenDecoder(CDVDStreamInfo &hints)
   // handle video rate
   if (hints.fpsrate > 0 && hints.fpsscale != 0)
   {
-    // then ffmpeg avg_frame_rate next
-    am_private->video_rate = 0.5 + (float)UNIT_FREQ * hints.fpsscale / hints.fpsrate;
+    // ensure the hints yield a sensible refresh rate
+    if (hints.fpsrate / hints.fpsscale <= 120)
+    {
+      // then ffmpeg avg_frame_rate next
+      am_private->video_rate = 0.5 + (float)UNIT_FREQ * hints.fpsscale / hints.fpsrate;
+    }
+  }
+
+  //if we had invalid fps hints then fallback to current refresh rate
+  if (!am_private->video_rate) {
```

Comment:
Kodi style? :-D

Reply:
Apologies for missing this. It now looks like my PR may need some re-work, so will ensure this is corrected as part of the re-write.

Reply:
Now resolved